src/SpeedDial/SpeedDial.js

```diff
@@ -0,0 +1,128 @@
+// @flow weak
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { withStyles } from 'material-ui/styles';
+import classNames from 'classnames';
+import Button from 'material-ui/Button';
+import Icon from 'material-ui/Icon';
+import AddIcon from 'material-ui-icons/Add';
+
+const styles = theme => ({
+  root: {
+    width: 56,
+    textAlign: 'center',
+    zIndex: theme.zIndex.speedDial,
+    transition: 'transform 0.1s ease-in-out',
+  },
+  hidden: {
+    transform: 'scale(0, 0)',
+  },
+  visible: {
+    transform: 'scale(1, 1)',
+  },
+  fabIcon: {
+    transition: 'transform 0.2s ease-in-out',
+  },
+  openFabIcon: {
+    transform: 'rotate(135deg)',
+  },
+  actions: {
+    position: 'absolute',
+  },
+  closedActions: {
+    transition: 'top 0s linear 0.2s',
+  },
+});
+
+function SpeedDial(props) {
+  const {
+    children: childrenProp,
+    classes,
+    className: classNameProp,
+    hidden,
+    icon: iconProp,
+    open,
+    onClick,
+    ...other
+  } = props;
+  const rootClassName = classNames(
+    classes.root,
+    hidden && classes.hidden,
+    hidden && classes.hidden,
+    classNameProp,
+  );
+  const fabIconClassName = classNames(classes.fabIcon, open && classes.openFabIcon);
+  const buttonsClassName = classNames(classes.actions, !open && classes.closedActions);
+
+  const children = React.Children.map(childrenProp, (child, childIndex) => {
+    if (!React.isValidElement(child)) return null;
+    const delay = 30 * (open ? childrenProp.length - childIndex : childIndex);
```

Comment:
Humm. I would rather be incrementing a counter to account for the null children that can be provided.

Reply:
Could you explain a bit?